test-interface-serialization/src/main/scala/scala/scalanative/testinterface/serialization/Serializer.scala

```diff
@@ -0,0 +1,189 @@
+package scala.scalanative
+package testinterface
+package serialization
+
+import scala.compat.Platform.EOL
+import sbt.testing._
+
+import scala.collection.mutable.ListBuffer
+
+object Serializer {
+  def serialize[T: Serializable](v: T): Iterator[String] =
+    implicitly[Serializable[T]].serialize(v)
+  def deserialize[T: Serializable](in: Iterator[String]): T =
+    implicitly[Serializable[T]].deserialize(in)
```

Comment:
Given that serialization here isn't meant to be extensible, it's better to go for lower-level abstraction-free approach here. Current implementation is probably rather slow as it relies on virtual dispatch heavily.

First of all type-classes are an overkill here due to the fact that all instances are statically known and we don't expect any extensibility beyond that point.

Second, for serialization it's best to have a mutable builder-like code to minimize I/O overhead. Please have a look at [NIR serialization](https://github.com/scala-native/scala-native/tree/master/nir/src/main/scala/scala/scalanative/nir/serialization) and [Show builder](https://github.com/scala-native/scala-native/blob/master/util/src/main/scala/scala/scalanative/util/Show.scala) for examples. We used to do high-level type-class like code and it was very expensive, even on the JVM (2-3x performance overhead.)